Source/VMware.vSphereDSC/DSCResources/VMHostAgentVM.ps1

```diff
@@ -0,0 +1,266 @@
+
+The BSD-2 license (the "License") set forth below applies to all parts of the Desired State Configuration Resources for VMware project.  You may not use this file except in compliance with the License.
+
+BSD-2 License
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+
+Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+[DscResource()]
+class VMHostAgentVM : VMHostBaseDSC {
+    <#
+    .DESCRIPTION
+
+    Specifies the Datastore used for deploying Agent VMs on this host.
+    #>
+    [DscProperty()]
+    [string] $AgentVmDatastore
+
+    <#
+    .DESCRIPTION
+
+    Specifies the Management Network for Agent VMs on this host.
+    #>
+    [DscProperty()]
+    [string] $AgentVmNetwork
+
+    hidden [string] $vCenterProductId = 'vpx'
+    hidden [string] $AgentVmDatastoreName = 'AgentVmDatastore'
+    hidden [string] $AgentVmNetworkName = 'AgentVmNetwork'
+    hidden [string] $GetAgentVmDatastoreAsViewObjectMethodName = 'GetAgentVmDatastoreAsViewObject'
+    hidden [string] $GetAgentVmNetworkAsViewObjectMethodName = 'GetAgentVmNetworkAsViewObject'
+
+    [void] Set() {
+        $this.ConnectVIServer()
+        $this.EnsureConnectionIsvCenter()
+        $vmHost = $this.GetVMHost()
+        $esxAgentHostManager = $this.GetEsxAgentHostManager($vmHost)
+
+        $this.UpdateAgentVMConfiguration($vmHost, $esxAgentHostManager)
+    }
+
+    [bool] Test() {
+        $this.ConnectVIServer()
+        $this.EnsureConnectionIsvCenter()
+        $vmHost = $this.GetVMHost()
+        $esxAgentHostManager = $this.GetEsxAgentHostManager($vmHost)
+
+        return !$this.ShouldUpdateAgentVMConfiguration($esxAgentHostManager)
+    }
+
+    [VMHostAgentVM] Get() {
+        $result = [VMHostAgentVM]::new()
+        $result.Server = $this.Server
+
+        $this.ConnectVIServer()
+        $this.EnsureConnectionIsvCenter()
```

Comment:
Are you sure it is only possible to configure those through the VC? Isnt' possible to configure directly on the esx?

Reply:
The EsxAgentHostManager is $null and it is not available for usage when the connection is to the ESXi directly.